### PR TITLE
Create popular categories block

### PR DESCRIPTION
**Pull request by DeLettenhove**
Create popular categories section on the student home page.

image

### CODE REVIEW

File: src/components/service-card/ServiceCard.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    backgroundColor: 'basic.white',
+    boxShadow: serviceCardShadow,
+    borderRadius: '6px',
+    '&:hover': {
+      boxShadow: serviceCardHoverShadow
+    }
+  }
+}
+
+const ServiceCard = ({ img, title, count, link }) => {
```
Comment (maarekua) [inline src/components/service-card/ServiceCard.js:21]: Weird name for the component.

File: src/components/service-card/ServiceCard.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <Button component={ Link } sx={ styles.card } to={ link }>
+      <Box
+        alt='Category image' component='img' src={ img }
+        sx={ { mr: '24px' } }
+      />
+      <Box sx={ { textAlign: 'start' } }>
```
Comment (maarekua) [inline src/components/service-card/ServiceCard.js:28]: title with description

Comment (DeLettenhove) [reply src/components/service-card/ServiceCard.js]: Fixed.

File: src/hooks/use-categories.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react'
+
+const useCategories = (items, count, step) => {
+  const [services, setServices] = useState(() => {
+    return items.length > count ? items.slice(0, count) : items
+  })
+  const [limit, setLimit] = useState(count + step)
+
```
Comment (maarekua) [inline src/hooks/use-categories.js:8]: const [items, setItems] = useState([allItems.slice(0, start)])

useEffect(() => {
const limit = items.lenght + step
setItems(allItems.slice(0, limit))
}, [allItems, start, step])

Comment (DeLettenhove) [reply src/hooks/use-categories.js]: Fixed.

File: src/components/service-card/ServiceCard.js (hunk 1 above)
Comment (DeLettenhove) [reply src/components/service-card/ServiceCard.js]: Fixed.

File: src/hooks/use-show-more.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const useShowMore = (allItems, start, step) => {
+  const [currentPage, setCurrentPage] = useState(0)
+  const [items, setItems] = useState([])
+
+  const totalPages = useMemo(() => {
+    return allItems.length > start ? Math.ceil((allItems.length - start) / step) : 0
+  }, [allItems, start, step])
+
+  const expandable = currentPage < totalPages
```
[the remark this answers is not in the document]
Comment (DeLettenhove) [reply src/hooks/use-show-more.js]: Fixed.

File: src/components/clickable-card/ClickableCard.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { Box } from '@mui/material'
+
+import { commonHoverShadow, commonShadow } from '~/styles/app-theme/custom-shadows'
```
[the remark this answers is not in the document]
Comment (DeLettenhove) [reply src/components/clickable-card/ClickableCard.js]: Fixed.

**Review by rostyslaw27 — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**